### ISSUE TITLE
make "all offline caches" mode available on map (in addition to "live")

### ISSUE THREAD

**Issue by ztNFny** [cgeo/cgeo#16360]
map in c:geo - at least the old one, for Unified i'm not 100% sure - has 3 modes:
- display only the selected object (caches, cache, zone, waypoint) (Transient)
- Live: Display all offline stored caches + query the online services
- Offline: Display all offline stored caches
"Offline" mode can be experienced by mapping a single cache, enabling live and disabling it again. Map will then show all offline stored caches.

I'd request the "live" button to be enhanced with a long-tap action to not just toggle live on/off but to explicitly enable "offline" mode. As this third map mode is not something that most users are even aware about I'd say "hiding" it behind a long-tap action would be fine.

Usecase:
- Circumstances: I do all my caching with offline data
- Goal: I'm mainly doing a multi but in case I walk past a cache on the route I of course want to find it.
- In the multi I map the next waypoint
- On map I enable "live" to see other caches as well
- This however now sends queries to the services, which a) cause traffic I don't want, b) load cache information I don't want, e.g. if there's a cache in my local list with corrected coords but online they aren't corrected then that cache jumps to the original coords.

**Comment by SammysHP** [comment]
It is either online, meaning that c:geo also queries online source in addition to locally stored caches, or offline, meaning that it won't send any network requests. What is shown in the offline mode depends on the situation:

- single cache: open the map for a single cache (and its waypoints), online is disabled.
- list: shows all caches of a list, online is disabled.
- live map with online disabled: this shows the result of the live map, but without asking the servers for more caches.

This means that "live/offline" is kind of a separate state not related to what is shown on the map. It simply enables or disable network requests.

Now, how to show all offline stored caches? Open the "All" list and show it on the map, or go to the live map and select the "All" list from the drop down list at the top.

**Comment by ztNFny** [comment]
> Now, how to show all offline stored caches? Open the "All" list and show it on the map, or go to the live map and select the "All" list from the drop down list at the top.

I know that this is a way - however that way doesn't match the usecase at all.
To use the wording you used in your list of the 3 possibilities (which afaict is just a rewording of mine) I want an option to enable "live map with online disabled" from all map views, in addition to the existing "live map with online enabled" aka Live button

**Comment by ztNFny** [comment]
> This means that "live/offline" is kind of a separate state not related to what is shown on the map. It simply enables or disable network requests.

That's not true as far as the GUI functionality (= button in map toolbar) is concerned. No matter which map view you're on, it always enables the display of all offline caches + enables network requests.

**Comment by moving-bits** [comment]
My typical workaround for this is: Map whatever you want to map (eg.: a single cache) to adjust the map's viewport, then use bottom navigation to toggle to cache list view (or any other screen), then back to map. It will have the same viewport as above, but will show all offline caches.

Implementing this long tap action seems useful to me, though we would need to think about how to implement it technically in a consistent way.

**Comment by ztNFny** [comment]
I hacked together "something" during the team call: https://github.com/ztNFny/cgeo/commit/91619596e54ceb456a11cb4276b19ac117d04869

Not nice, not production ready, no proper design - but working ;-)

**Comment by ztNFny** [comment]
Cleaned this one up a bit: https://github.com/cgeo/cgeo/compare/master...ztNFny:cgeo:mapmode

**Comment by ztNFny** [comment]
Cleaned this one up a bit: https://github.com/cgeo/cgeo/compare/master...ztNFny:cgeo:mapmode
It's using a setting as I couldn't find a suitable place accessible from both classes, but that's the only "ugly" thing left in there.

**Comment by ztNFny** [comment]
> Cleaned this one up a bit: [master...ztNFny:cgeo:mapmode](https://github.com/cgeo/cgeo/compare/master...ztNFny:cgeo:mapmode) It's using a setting as I couldn't find a suitable place accessible from both classes, but that's the only "ugly" thing left in there.

rebase to latest master pushed
the latest update to live map made it easier to enable this mode by enabling and disabling live map though, so not sure whether this is still worth it?

**Comment by ztNFny** [comment]
A way simplified approach, utilizing the fact that there's now a delay before the live mode request is actually sent to the server (would conflict with https://github.com/cgeo/cgeo/issues/16566). It just enables and then disables live map mode.
https://github.com/cgeo/cgeo/compare/master...ztNFny:cgeo:mapmode_simple

**Comment by eddiemuc** [comment]
Due to https://github.com/cgeo/cgeo/issues/16566#issuecomment-2591201459 I read this issue and I am confused.

If I understand you want the same behavior as "Live map", only that online servers are never called, only the database is queried as you move your viewport.

Is this not what happens if you enter "normal" map (via bottom navigation) with "live" button disabled?

**Comment by eddiemuc** [comment]
> Is this not what happens if you enter "normal" map (via bottom navigation) with "live" button disabled? 

And if it currently isn't: shouldn't it be that way?

**Comment by ztNFny** [comment]
> Is this not what happens if you enter "normal" map (via bottom navigation) with "live" button disabled?

There currently is no way to enter "normal map with live disabled" (unless you happend to use map with live disabled before in which case you're *returning* to it).
This request and the related commits are about adding just that: Long-tap on the live button to turn "any offline map" (e.g. a list map, single cache map) into "database map"

**Comment by ztNFny** [comment]
> A way simplified approach, utilizing the fact that there's now a delay before the live mode request is actually sent to the server (would conflict with #16566). It just enables and then disables live map mode.
> [master...ztNFny:cgeo:mapmode_simple](https://github.com/cgeo/cgeo/compare/master...ztNFny:cgeo:mapmode_simple)

rebased that onto the latest master, this approach luckily still works, live map is disabled quick enough before a network request is sent. So I'll open a PR for this one.

**Comment by eddiemuc** [comment]
> There currently is no way to enter "normal map with live disabled" (unless you happend to use map with live disabled before 

I am still not getting why there's a change needed:
- in case you want to cache offline you just have to enter the map via bottom navigation and tap the "live" button one time (to turn off live data). Then database caches will be shown while you browse the map. but no online query is made
- if "live" was on but you tap live button within "waiting time", no query will ever be send to servers
- c:geo even remembers this decision and will open "offline" already on subsequent calls.

What am I missing?

**Comment by ztNFny** [comment]
> What am I missing?

It's not a new functionality, it's just making it more accessible by replacing multiple clicks with 1 long-click

**Comment by eddiemuc** [comment]
But if I am correct with my guessing then it's already only one click

**Comment by ztNFny** [comment]
> But if I am correct with my guessing then it's already only one click

It's only 1 click if you're a) not on a map view but another view with bottom nav visible + b) you left live map in non-live mode the last time.

**Comment by eddiemuc** [comment]
... zero clicks if you turned live off before

...and Double-click-on-one-button if you start from a Single-Cache/list view

**Comment by bekuno** [comment]
... == not intuitive

I believe that 95% do not know this posibility.

**Comment by eddiemuc** [comment]
> ... == not intuitive
> 
> I believe that 95% do not know this posibility.

True. But the same 95% will not know about the long-tap-option either. I still don't understand why a double -tap should be less intuitive than a long-tap. Also, this is only relevant for the scenario where one is coming from a mapped single cache/waypoint/list.

**Comment by bekuno** [comment]
> > ... == not intuitive
> > I believe that 95% do not know this posibility.
> 
> True. But the same 95% will not know about the long-tap-option either.

That is also true.

To have two different functions on one button feels bad to me.

**Comment by moving-bits** [comment]
Implemented with #16753

**Comment by murggel** [comment]
And it's already documented in manual 
https://manual.cgeo.org/en/hiddenfunctions
and
https://manual.cgeo.org/en/mainmenu/livemap#top_bar_menu

**Comment by moving-bits** [comment]
> And it's already documented in manual [manual.cgeo.org/en/hiddenfunctions](https://manual.cgeo.org/en/hiddenfunctions)

BTW: this is exactly the place where you could add the hidden settings for actionbar coloring
